app/Http/Controllers/MergeController.php

```
@@ -53,17 +60,39 @@ public function store($id, Request $request)
         $duplicateFields = array_except($duplicate->toArray(), $metadata);
         $duplicateFieldNames = array_keys($duplicateFields);
 
+        // Find out which fields we need to handle merging
+        $intersectedFields = array_intersect_key($target->toArray(), array_flip($duplicateFieldNames));
+
+        // Fields that we can automatically merge
+        $fieldsToMerge = array_except($duplicateFields, array_keys($intersectedFields));
         // Are there fields we can't automatically merge? Throw an error.
         if (count(array_intersect_key($target->toArray(), array_flip($duplicateFieldNames)))) {
-            $errors = collect($duplicateFieldNames)->map(function ($fieldName) {
-                return 'Cannot merge "'.$fieldName.'" into non-null field on target.';
-            });
-
-            throw new NorthstarValidationException($errors, ['target' => $target, 'duplicate' => $duplicate]);
+            $unmergedFields = [];
+
+            // Call merge on intersecting fields
+            foreach ($intersectedFields as $field => $value) {
+                $merged = $this->merger->merge($field, $target->$field, $duplicate->$field);
```

I've went in circles thinking about this, so here's some unstructured thoughts:

I'm wary about using `$merged` to track both the _new_ value and an "unmergeable field" is what we do when we have merged values that are false (e.g. `sms_paused`). Right now if the Merger's mergeSmsPaused` method returned false, it'd end up in the "unmerged" array. Instead, let's throw an exception when something goes wrong.

We might want to let the Merger handle assigning the new values. For example, we'll probably want to "bundle" `sms_status` and `sms_paused` with whichever account gets the `mobile` field... and actually, speaking of, should we be resetting those to `null` if a user's mobile number changes?

This might be a good thing to huddle on IRL.